clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -198,6 +196,73 @@ public void testExists_NotExists() throws ApiException, IOException {
 
         Assert.assertFalse(fs.exists(p));
     }
+    
+    @Test
+    public void testDelete_FileExists() throws ApiException, IOException {
+        when(objectsApi.statObject(eq("repo"), eq("main"), eq("delete/sample/file.txt")))
+                .thenReturn(new ObjectStats().
+                        path("lakefs://repo/main/delete/sample/file.txt").
+                        pathType(PathTypeEnum.OBJECT).
+                        physicalAddress(s3Url("delete/sample/file.txt")).
+                        checksum("unused").
+                        mtime(0L).
+                        sizeBytes(42L));
+        // return true because file found
+        Assert.assertTrue(
+                fs.delete(new Path("lakefs://repo/main/no/place/file.txt"), false));
```

Comment:
This tests metadata mutation but not data mutation.  If you created a file and passed it in the physical address, you could verify it was actually deleted from minio.

Reply:
Discussed this idea with @talSofer - we test the side-effect of the filesystem and not testing lakefs.
Currently delete is a request to lakeFS metadata only. We do not delete the data file. This is why we also do not check if the physical file was deleted.
Checking if the file is deleted means we check lakeFS.

Reply:
Oh right, I forgot.  You are (of course) 100% correct (and I am the corresponding percentage _less_ than 100% correct...).